### PR TITLE
Selective Sync: Open sub folder context menu #5596

### PR DESCRIPTION
**Pull request by guruz**

### CODE REVIEW

File: src/gui/accountsettings.cpp [hunk 1]
```diff
@@ -403,6 +426,16 @@ void AccountSettings::slotOpenCurrentFolder()
     }
 }
 
+void AccountSettings::slotOpenCurrentLocalSubFolder()
+{
+    QModelIndex selected = ui->_folderList->selectionModel()->currentIndex();
+    if( !selected.isValid() )
```
Comment (ckamm) [inline src/gui/accountsettings.cpp]: Add a check for `_model->classify(indx) == FolderStatusModel::SubFolder`?

File: src/gui/accountsettings.cpp [hunk 2]
```diff
@@ -403,6 +426,16 @@ void AccountSettings::slotOpenCurrentFolder()
     }
 }
 
+void AccountSettings::slotOpenCurrentLocalSubFolder()
+{
+    QModelIndex selected = ui->_folderList->selectionModel()->currentIndex();
```
Comment (ckamm) [inline src/gui/accountsettings.cpp]: I'm slightly worried that the selected item could be different from the one that was clicked on. I couldn't make that happen in my tests though.

**Comment by ckamm** [comment]
Looks good and works in my tests, added two minor comments.

**Comment by ogoffart** [comment]
You should move your code in AccountSettings::slotCustomContextMenuRequested  rather than in AccountSettings::slotFolderListMousePressed.  That way, it also work with the "open menu" key on the keyboard. (Accessibility)

File: src/gui/accountsettings.cpp (hunk 2 above)
Comment (ogoffart) [inline src/gui/accountsettings.cpp]: I think that's fine.

Altough there is no need to go trough the selectionModel:  `ui->_folderList->currentIndex()` should work

File: src/gui/accountsettings.cpp (hunk 1 above)
Comment (ogoffart) [inline src/gui/accountsettings.cpp]: Normally, that function should only be called in this case. But it's true we could play it save.

File: src/gui/accountsettings.h [hunk 3]
```diff
@@ -75,7 +75,8 @@ protected slots:
     void slotScheduleCurrentFolderForceRemoteDiscovery();
     void slotForceSyncCurrentFolder();
     void slotRemoveCurrentFolder();
-    void slotOpenCurrentFolder();
+    void slotOpenCurrentFolder(); // sync folder
+    void slotOpenCurrentLocalSubFolder(); // selected subfolder in sync folder
```
Comment (ogoffart) [inline src/gui/accountsettings.h]: (Maybe the two function could be merged in one)

**Comment by guruz** [comment]
@ckamm @ogoffart Please re-review.
I didn't see/know about a context menu signal even..

**Comment by ogoffart** [comment]
:+1:

**Pull request merged**